var/lib/delphix-platform/ansible/apply

```diff
@@ -21,6 +21,13 @@ SOURCE_DIRECTORY="$(readlink -f "${BASH_SOURCE%/*}")"
 DLPX_ANSIBLE_DIRECTORY="${DLPX_ANSIBLE_DIRECTORY:-$SOURCE_DIRECTORY}"
 ROOT_FILESYSTEM=$(zfs list -Ho name /)
 
+#
+# If we are running as part of live-build, the root of the appliance's
+# filesystem won't be /
+#
+APPLIANCE_ROOT_DIR="$SOURCE_DIRECTORY/../../../.."
```

Comment:
I was talking with Prakash about this yesterday and I wasn't sure if we needed to have a more global concept of the ROOT for the appliance build. Also this probably needs to be inside an if-block and only execute when we $DLPX_ANSIBLE_CONNECTION == chroot.

Reply:
Ok, when we do upgrade, I wasn't sure if it's called from the same level in the directory tree. If you've verified this then we wouldn't need to guard this with `chroot` check.

Reply:
Would it be better to expose this as an environment variable? Then appliance-build can pass in that information via the variable, and if it's not provided it can default to `/` (or perhaps the service can also provide it, and this script would error if it's empty)?

This just strikes me as hack, and easily broken if the location of this script changes.

Thoughts? I'm OK with this if folks would rather do this, but just wanted to bring up another approach.

This also doesn't solve the issue for the `zfs list /` call above, but I'm not sure we can fix that will the same solution (although we should probably use the same "appliance root" in the `zfs list`, even if it still doesn't work).

Reply:
While I don't love the change that I proposed, I also don't love adding another environment variable that has to be set, since the caller of this script lives in another repository. I think it would be good to keep the interface between the two repositories as simple where possible to prevent flag days.

I think ideally we would enter the chroot before calling the `apply` script, so that it doesn't need to be aware of the difference scenarios in which it can be called. However, I know that Ansible does need to be aware when it is running in a chroot, so it would need to be able to detect that itself. I'm not sure whether that's possible.

Reply:
Yea... the fact that this script is called outside of the chroot is to hack around the fact that ansible can't be called directly from the chroot.

OK, I'm good with this solution. I can't think of anything better at the moment.

Reply:
I agree. It's hacky but can live with this until we have a broader need to make it more robust.